### ISSUE TITLE
Predefined variables with JWT token and @user

### ISSUE THREAD

**Issue by ADCDS** [SoftInstigate/restheart#530]
**Describe the bug**

When setting up a change stream in RESTHeart to filter documents based on the authenticated user's ID from a JWT token, the `@user.userId` variable is not being substituted correctly within the `$var` operator in the stream definition.

**To Reproduce**

1.  **Create a collection** named `secrets`, with documents containing an `author` field.

2. Setup JWT and disable mongoRealmAuthenticator in restheart.yml: 
    `# Disable mongoRealmAuthenticator
    mongoRealmAuthenticator:
      enabled: false
    
    jwtAuthenticationMechanism:
      enabled: true
      algorithm: HS256
      key: secret
      usernameClaim: sub
      rolesClaim: roles
      base64Encoded: false
      account-properties-claims:
        - userId`

3.  **Define the streams** :

    `{ "stages" : [
          { "_$match" : { "fullDocument::author" : "@user.userId" } }
        ],
        "uri" : "mine"
      }`

4.  **Define the streams ACL**:
    `{
      "_id": "allowWebSocketConnection",
      "roles": [
        "user"
      ],
      "predicate": "path-prefix('/secrets/_streams/mine')",
      "priority": 100,
      "_etag": {
        "$oid": "6724fa1fb899c94afe99519b"
      }
    }`

6.  **Attempt to connect** to the `mine` stream via WebSocket:

    `websocat --text -H='Authorization: Bearer [JWT_TOKEN]' - ws://127.0.0.1:8080/secrets/_streams/mine`
With JWT_TOKEN being something like this: `eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiIxMjM0NTY3ODkwIiwibmFtZSI6IkpvaG4gRG9lIiwiaWF0IjoxNTE2MjM5MDIyLCJyb2xlcyI6WyJ1c2VyIl0sInVzZXJJZCI6ImJvYiJ9.6DFKnQ2Lnzi9ozCMMf-yOFyi4ZJ61qsaSIUiHAFe96w`

7.  **Observe the error**:

    -   The server responds with a `400 Bad Request`.
    -   Server logs display:

        `Cannot open change stream, the request does not specify the required variables in the avars query parameter: variable @user.userId not bound`

**Expected behavior**

The variable `@user.userId` should be substituted with the `userId` from the JWT token within the `$var` operator. This would allow users to receive change stream updates only for documents where the `author` field matches their own `userId`.

**Comment by ujibang** [comment]
The predefined variables were not injected for ChangeStreams (your test aggregation definition was working if used to defined an aggregation rather than a change stream).

This is fixed by commit dd448f564d0d34c45186834b3eb057f8b053363f

@ADCDS can you please try with snapshot build

```bash
$ docker pull softinstigate/restheart-snapshot:dd448f5
$ docker run -e RHO="/jwtAuthenticationMechanism/enabled-> true;/http-listener/host->'0.0.0.0';/mclient/connection-string->'mongodb://host.docker.internal'" -p 8080:8080 softinstigate/restheart-snapshot:dd448f5
```

now you can define the change stream

```bash
$ echo '{"_id": "_properties.secrets",
  "streams": [
    {
      "stages": [
        {
          "_$match": {
            "_$or": [
              {
                "operationType": "insert"
              },
              {
                "operationType": "update"
              }
            ]
          }
        }
      ],
      "uri": "all"
    },
    {
      "stages": [
        {
          "_$match": {
            "fullDocument::author": {
              "$var": "@user.userId"
            }
          }
        }
      ],
      "uri": "mine"
    }
  ]
}' |  http -a admin:secret patch :8080/coll
```

and finally connect to the change stream

```bash
$ websocat -U -n -H='Authorization: Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiIxMjM0NTY3ODkwIiwibmFtZSI6IkpvaG4gRG9lIiwiaWF0IjoxNTE2MjM5MDIyLCJ1c2VySWQiOiJhZG1pbiIsInJvbGVzIjpbImFkbWluIl0sImlzcyI6InJlc3RoZWFydC5vcmcifQ.H9dvFzyBzwxYswUkjMahAmCmmugDSRsJGluxBkrLv5c' --text ws://127.0.0.1:8080/coll/_streams/mine
```

**Comment by ADCDS** [comment]
It worked!
Thank you very much!
